packages/ui-components/src/hooks/useTableSorting.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { useCallback, useState } from 'react';
+import { getSortByKey } from '@tupaia/utils';
+
+const DIRECTION = {
+  ASC: 'asc',
+  DESC: 'desc',
+};
+
+const toggleOrder = order => (order === DIRECTION.DESC ? DIRECTION.ASC : DIRECTION.DESC);
+
+/**
+ * @param {string | undefined} initialOrderBy (Initial) key of a sorted column
+ * @param {'asc' | 'desc'} initialOrder
+ */
+export const useTableSorting = (data, initialOrderBy, initialOrder = DIRECTION.ASC) => {
```

Comment:
I placed this under a new folder called `hooks`. Not sure if this is better than placing this close to the component it's related to (`Table`). I thought it may be good for hooks and components to be defined in separate locations, to encourage less coupling between them

Reply:
Nice - good move. I can image there will be more hooks to add over time!